test/javascripts/vendor/promise-polyfill/promise.js

```diff
@@ -0,0 +1,233 @@
+(function (root) {
```

Comment:
The rationale behind not using NPM was:
- This is a critical fix that I was trying to get out in a day
- You said something about NPM imports not always working when I asked about using NPM

In an ideal world, we'd be using NPM to import promise + modern versions sinon, chai, etc.